### PR TITLE
Separate the payment info in a separate method and use it in show_pay…

### PR DESCRIPTION
**Pull request by RafyAmgadBenjamin**
### Description
- Show the payment information during waiting for the payment screen in the marketplace solutions.
- Update the message displayed in the payment screen

Describe the changes introduced by this PR and what does it affect

### Changes
Separate the payment into a single method to be called  `wait_pool_payment` method and  `show_payment`

### Issue 
https://github.com/threefoldtech/js-sdk/issues/1027
https://github.com/threefoldtech/js-sdk/issues/1026

### CODE REVIEW

File: jumpscale/sals/reservation_chatflow/deployer.py [hunk 1]
```diff
@@ -1545,5 +1552,27 @@ def wait_pool_payment(self, bot, pool_id, exp=5, qr_code=None, trigger_cus=0, tr
...
+        total_amount = escrow_info.amount
+        total_amount_dec = Decimal(total_amount) / Decimal(1e7)
+        thecurrency = escrow_asset.split(":")[0]
+        total_amount = "{0:f}".format(total_amount_dec)
+        qr_code = f"{thecurrency}:{escrow_address}?amount={total_amount}&message=p-{resv_id}&sender=me"
+        msg_text = f"""
+        
+        <h4> Wallet Address: </h4>  {escrow_address} \n
+        <h4> Currency: </h4>  {thecurrency} \n
+        <h4> Memo Text (Reservation Id): </h4>  p-{resv_id} \n
```
Comment (xmonader) [inline jumpscale/sals/reservation_chatflow/deployer.py:1569]: Can we please make it Reservation ID

File: jumpscale/sals/reservation_chatflow/deployer.py [hunk 2]
```diff
@@ -1545,5 +1552,27 @@ def wait_pool_payment(self, bot, pool_id, exp=5, qr_code=None, trigger_cus=0, tr
...
+        escrow_address = escrow_info.address
+        escrow_asset = escrow_info.asset
+        total_amount = escrow_info.amount
+        total_amount_dec = Decimal(total_amount) / Decimal(1e7)
+        thecurrency = escrow_asset.split(":")[0]
+        total_amount = "{0:f}".format(total_amount_dec)
+        qr_code = f"{thecurrency}:{escrow_address}?amount={total_amount}&message=p-{resv_id}&sender=me"
+        msg_text = f"""
+        
+        <h4> Wallet Address: </h4>  {escrow_address} \n
```
Comment (xmonader) [inline jumpscale/sals/reservation_chatflow/deployer.py:1567]: Can we make it `Destination Wallet Address` to be consistent with screenns

File: jumpscale/sals/reservation_chatflow/deployer.py [hunk 3]
```diff
@@ -1545,5 +1552,27 @@ def wait_pool_payment(self, bot, pool_id, exp=5, qr_code=None, trigger_cus=0, tr
...
+        <h4> Currency: </h4>  {thecurrency} \n
+        <h4> Memo Text (Reservation Id): </h4>  p-{resv_id} \n
+        <h4> Total Amount: </h4> {total_amount} {thecurrency} \n
+
+        <h5>Inserting the memo-text is an important way to identify a transaction recipient beyond a wallet address. Failure to do so will result in a failed payment. Please also keep in mind that an additional Transaction fee of 0.1 FreeTFT will automatically occurs per transaction.</h5>
```
Comment (m-motawea) [inline jumpscale/sals/reservation_chatflow/deployer.py:1572]: seems you have 0.1 FreeTFT hardcoded in the fee part. should be 0.1 {thecurrency}

File: jumpscale/sals/reservation_chatflow/deployer.py [hunk 4]
```diff
@@ -1545,5 +1552,27 @@ def wait_pool_payment(self, bot, pool_id, exp=5, qr_code=None, trigger_cus=0, tr
...
+        total_amount = "{0:f}".format(total_amount_dec)
+        qr_code = f"{thecurrency}:{escrow_address}?amount={total_amount}&message=p-{resv_id}&sender=me"
+        msg_text = f"""
+        
+        <h4> Destination Wallet Address: </h4>  {escrow_address} \n
+        <h4> Currency: </h4>  {thecurrency} \n
+        <h4> Memo Text (Reservation ID): </h4>  p-{resv_id} \n
+        <h4> Total Amount: </h4> {total_amount} {thecurrency} \n
+
+        <h5>Inserting the memo-text is an important way to identify a transaction recipient beyond a wallet address. Failure to do so will result in a failed payment. Please also keep in mind that an additional Transaction fee of 0.1 {thecurrency} will automatically occurs per transaction.</h5>
```
Comment (waleedhammam) [inline jumpscale/sals/reservation_chatflow/deployer.py:1572]: we can align the string here as we use dedent now, if can fix it, just will make the code looks prettier

**Review by ranatrk — APPROVED** [review]

**Review by waleedhammam — APPROVED** [review]

**Pull request merged**